src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export interface Figure {
+  shape:Shape;
+  color:Color;
 
+  getArea():number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color:Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    const isSideCorrect = (...args:number[]):boolean => {
+      return args.some((arg:number) => {
+        return arg <= 0;
+      });
+    };
+
+    if (isSideCorrect()) {
+      throw new Error('Incorrect side length parameter');
+    }
+
+    if ((a >= b + c) || (b >= a + c) || (c >= a + b)) {
+      throw new Error(`${a}, ${b} and ${c} can't be a triangle`);
+    }
+  }
 
+  getArea():number {
+    const getHalf: number = (this.a + this.b + this.c) / 2;
+
+    return Number(Math.sqrt(
+      getHalf * (getHalf - this.a) * (getHalf - this.b) * (getHalf - this.c),
+    ).toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Incorrect radius. Radius must be greater then zero');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * (this.radius ** 2) * 100) / 100;
```

Comment:
Optional: you round twice over the code(43-46 lines too). You also could create a separate helper function - `rounder` for example for this purpose and round it using one way using - `Math.floor or toFixed()`.